### PR TITLE
[FIX] when specifying a bind, if the consumer didn't specify deliver_to get flagged as pull

### PR DESCRIPTION
**Pull request by aricart**

### CODE REVIEW

**Review by caspervonb — COMMENTED** [review]
lgtm

File: nats-base-client/jsclient.ts [hunk 1]
```diff
@@ -330,7 +330,7 @@ export class JetStreamClientImpl extends BaseApiClient
     const cso = await this._processOptions(subject, opts);
     // this effectively requires deliver subject to be specified
     // as an option otherwise we have a pull consumer
-    if (!cso.config.deliver_subject) {
+    if (!cso.isBind && !cso.config.deliver_subject) {
       throw new Error(
         "consumer info specifies a pull consumer - deliver_subject is required",
```
Comment (Jarema) [inline nats-base-client/jsclient.ts:335]:
```suggestion
        "deliver_subject is required for Push Consumer",
```

**Review by Jarema — CHANGES_REQUESTED** [review]
One thing - this error is very confusing.

I suggest improving it a little.

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**